public/app/plugins/panel/radialbar/suggestions.ts

```diff
@@ -1,101 +1,82 @@
-import { VisualizationSuggestionsBuilder } from '@grafana/data';
-import { FieldColorModeId } from '@grafana/schema';
+import { defaultsDeep } from 'lodash';
+
+import { FieldColorModeId, FieldType, VisualizationSuggestion, VisualizationSuggestionsHandler } from '@grafana/data';
+import { t } from '@grafana/i18n';
 import { GraphFieldConfig } from '@grafana/ui';
-import { SuggestionName } from 'app/types/suggestions';
 
 import { Options } from './panelcfg.gen';
 
-export class GaugeSuggestionsSupplier {
-  getSuggestionsForData(builder: VisualizationSuggestionsBuilder) {
-    const { dataSummary } = builder;
-
-    if (!dataSummary.hasData || !dataSummary.hasNumberField) {
-      return;
-    }
+export const radialBarSuggestionsHandler: VisualizationSuggestionsHandler<Options, GraphFieldConfig> = (
+  dataSummary
+) => {
+  if (!dataSummary.hasData || !dataSummary.hasFieldType(FieldType.number)) {
+    return;
+  }
 
-    // for many fields / series this is probably not a good fit
-    if (dataSummary.numberFieldCount >= 10) {
-      return;
-    }
+  // for many fields / series this is probably not a good fit
+  if (dataSummary.fieldCountByType(FieldType.number) >= 10) {
+    return;
+  }
 
-    const list = builder.getListAppender<Options, GraphFieldConfig>({
-      name: SuggestionName.Gauge,
-      pluginId: 'gauge',
-      options: {},
+  const withDefaults = (
+    suggestion: VisualizationSuggestion<Options, GraphFieldConfig>
+  ): VisualizationSuggestion<Options, GraphFieldConfig> => {
+    // if there is a string field and there are few enough rows, we assume it's tabular data and not numeric series data,
+    // and the de-aggregated version of the viz probably makes more sense
+    const isTabularData =
+      dataSummary.hasFieldType(FieldType.string) && dataSummary.frameCount === 1 && dataSummary.rowCountTotal < 10;
+    return defaultsDeep(suggestion, {
```

Comment:
having this handled by the list appender is a bit lost here, now each plugin author has to remember this, and build a function like withDefaults. 

Maybe we can pass in the data summary and the list appender as separate args to the supplier interface?

Reply:
but I guess that does not really solve it as the defaults needs to be supplied by the plugin like here (for previewModifier etc)